.github/workflows/ci-cd.yml

```
@@ -7,139 +7,44 @@ on:
-          echo "Next.js build directory:"
-          ls -la .next/ || echo ".next directory not found"
-          
+      
+      # Docker build only on main branch pushes
       - name: Set up Docker Buildx
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
         uses: docker/setup-buildx-action@v2
+      
       - name: Build Docker image
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
         uses: docker/build-push-action@v4
```

actionlint (1.7.4)

the runner of "docker/build-push-action@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)